### DISCUSSION TITLE
Публичный Docker Registry с образом werf

### DISCUSSION THREAD

**Discussion by AkakievKD** [werf/werf#4423] [General]
У вас есть активный докер реджестри с образом ?

**Comment by ilya-lesikov** [comment]
У нас есть образы для запуска werf через экспериментальный Buildah-бэкенд, без использования Docker'а: [список образов](https://ru.werf.io/documentation/v1.2/advanced/ci_cd/run_in_container/how_it_works.html#%D0%B4%D0%BE%D1%81%D1%82%D1%83%D0%BF%D0%BD%D1%8B%D0%B5-%D0%BE%D0%B1%D1%80%D0%B0%D0%B7%D1%8B-werf). В конечном итоге это будет единственный, рекомендуемый способ запуска.

Для запуска через Docker инструкции [здесь](https://ru.werf.io/documentation/v1.2/advanced/ci_cd/run_in_container/use_docker_container.html#%D1%81%D0%B1%D0%BE%D1%80%D0%BA%D0%B0-%D0%BE%D0%B1%D1%80%D0%B0%D0%B7%D0%BE%D0%B2-%D1%81-%D0%BF%D0%BE%D0%BC%D0%BE%D1%89%D1%8C%D1%8E-docker-%D1%81%D0%B5%D1%80%D0%B2%D0%B5%D1%80%D0%B0-%D0%BD%D0%B5-%D1%80%D0%B5%D0%BA%D0%BE%D0%BC%D0%B5%D0%BD%D0%B4%D1%83%D0%B5%D1%82%D1%81%D1%8F), но собрать образ потребуется самостоятельно.

**Reply by AkakievKD** [reply]
собираете слои через инструмент типо каники ?

**Reply by ilya-lesikov** [reply]
Да, что-то типа того. Используем редхатовский Buildah как go-библиотеку: https://github.com/containers/buildah

**Reply by AkakievKD** [reply]
Buildah как go-библиотеку: https://github.com/containers/buildah     он как компонент у вас для собственного инструмента или вы его как полноценный используете ?

**Reply by AkakievKD** [reply]
(У нас есть образы для запуска werf через экспериментальный Buildah-бэкенд) не совсем понял как это работает

**Reply by AkakievKD** [reply]
> Buildah как go-библиотеку: https://github.com/containers/buildah он как компонент у вас для собственного инструмента или вы его как полноценный используете ?

нашел ответ https://itnan.ru/post.php?c=1&p=649551

**Reply by ilya-lesikov** [reply]
Тут пример, как запустить werf через Buildah в K8s: https://ru.werf.io/documentation/v1.2/advanced/ci_cd/run_in_container/use_kubernetes.html
